rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -35,41 +36,37 @@
 
 namespace rosbag2_storage_plugins
 {
-
-SqliteStorage::SqliteStorage()
-: database_(),
-  write_statement_(nullptr),
-  read_statement_(nullptr),
-  message_result_(nullptr),
-  current_message_row_(nullptr, SqliteStatementWrapper::QueryResult<>::Iterator::POSITION_END)
-{}
-
 void SqliteStorage::open(
   const std::string & uri, rosbag2_storage::storage_interfaces::IOFlag io_flag)
 {
   auto metadata = is_read_only(io_flag) ?
     load_metadata(uri) :
     std::unique_ptr<rosbag2_storage::BagMetadata>();
 
+  std::string database_name;
+
   if (metadata) {
     if (metadata->relative_file_paths.empty()) {
       throw std::runtime_error(
               "Failed to read from bag '" + uri + "': Missing database file path in metadata");
     }
 
-    database_name_ = metadata->relative_file_paths[0];
+    database_name = metadata->relative_file_paths[0];
```

Comment:
I think this is an issue in the current implementation of rosbag. The differentiation between whether or not a bagfile has metadata should not happen in the actual storage plugin but in the rosbag Reader API. 

The way I see this is that the Reader API should verify if the uri given by the user points to a directory with a `metadata.yaml` file or is a filepath to a storage. 
If it's a metadata file, the reader should be in charge of iterating through the individual `relative_file_paths` and open the storages accordingly.

I think the `uri` argument passed to this `open` function should be unambiguously point to a filepath the storage plugin can open - in the case of SQLite it should be `*.db3` file.